### PR TITLE
Add start_slot parameter

### PR DESCRIPTION
**Pull request by danaugrs**
A first attempt to fix https://github.com/gfx-rs/wgpu/issues/215.

### CODE REVIEW

File: wgpu-native/src/command/render.rs [hunk 1]
```diff
@@ -336,7 +337,7 @@ pub extern "C" fn wgpu_render_pass_set_vertex_buffers(
             .buffers
             .use_extend(&*buffer_guard, id, (), BufferUsage::VERTEX)
             .unwrap();
-        vbs.total_size = buffer.size - offset;
+        vbs.total_size = buffer.size - (offset + start_slot as u64);
```
Comment (kvark) [inline wgpu-native/src/command/render.rs]: The starting slot is not in any way related to the buffer offset. it affects the position in `pass.vertex_state.inputs` that we start in.

Comment (danaugrs) [reply wgpu-native/src/command/render.rs]: Oh I see. Pushed a new version.

File: wgpu-native/src/command/render.rs [hunk 2]
```diff
@@ -338,8 +339,10 @@ pub extern "C" fn wgpu_render_pass_set_vertex_buffers(
             .unwrap();
         vbs.total_size = buffer.size - offset;
     }
-    for vbs in pass.vertex_state.inputs[length..].iter_mut() {
```
Comment (kvark) [inline wgpu-native/src/command/render.rs]: we should be using `[start_slot as usize + length ..]` here

File: wgpu-native/src/command/render.rs [hunk 3]
```diff
@@ -320,6 +320,7 @@ pub extern "C" fn wgpu_render_pass_set_index_buffer(
 #[no_mangle]
 pub extern "C" fn wgpu_render_pass_set_vertex_buffers(
     pass_id: RenderPassId,
+    start_slot: u32,
```
Comment (kvark) [inline wgpu-native/src/command/render.rs]: we still need to make it affect the `pass.vertex_state.inputs.iter_mut()` part, i.e. by doing `pass.vertex_state.inputs[start_slot..].iter_mut()`

File: wgpu-native/src/command/render.rs (hunk 2 above)
Comment (kvark) [reply wgpu-native/src/command/render.rs]: actually, now that we are changing this function to specify a subset of vertex buffers, I think this whole block needs to be removed (sorry for not noticing earlier!)

Comment (danaugrs) [reply wgpu-native/src/command/render.rs]: No prob - removed it.

**Pull request merged**